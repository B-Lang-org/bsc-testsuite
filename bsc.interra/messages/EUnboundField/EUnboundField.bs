-----------------------------------------------------------------------
-- Project: Bluespec

-- File: EUnboundField.bs

-- Author : Amit Grover      <amit@noida.interrasystems.com>

-- Description: This testcase triggers the EUnboundField error of the bluespec
-- compiler (Unbound Field)
--
-----------------------------------------------------------------------



package EUnboundField () where

data My_pair = Cons1 {fst :: Integer; snd :: Integer}


firstpair :: My_pair 
firstpair = Cons1 {fst =5; snd =10}

secondpair :: My_pair
secondpair = firstpair {fst=5}

