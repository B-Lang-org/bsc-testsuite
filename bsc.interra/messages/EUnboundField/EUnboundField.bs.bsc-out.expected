checking package dependencies
compiling EUnboundField.bs
Error: "EUnboundField.bs", line 24, column 24: (T0138)
  The expression does not have a field `fst' because it is not a struct or
  interface. Found type:
    EUnboundField.My_pair
